src/pages/experiments/[experimentId]/plots-and-tables/batch-differential-expression/index.jsx

```diff
@@ -56,21 +54,13 @@ const BatchDiffExpression = (props) => {
   const [comparison, setComparison] = useState(comparisonInitialState);
   const batchCellSetKeys = useSelector(getCellSetsHierarchyByKeys([comparison.basis]))[0]?.children
     .map((child) => child.key);
+  const isDatasetUnisample = (useSelector((state) => (
+    state.experimentSettings.info.sampleIds.length)) === 1);
 
   useEffect(() => {
     dispatch(loadCellSets(experimentId));
   }, []);
 
-  useEffect(() => {
-    if (hierarchy && hierarchy.length === 0) return;
-
-    const samples = hierarchy?.find(
-      (rootNode) => (rootNode.key === 'sample'),
-    )?.children;
-
-    setNumSamples(samples.length);
-  }, [Object.keys(properties).length]);
-
```

Comment:
This alternative implementation makes batch differential expression work for Seurat object uploads. The issue is that `state.experimentSettings.info.sampleIds.length` will always be 1 for Seurat object uploads (a single .rds file is uploaded with no way to know that there are multiple samples until after being processed).

The alternative uses the sample cellsets to determine the number of samples which will be correct for Seurat object uploads. It is also consistent with the current implementation is Data Exploration where there is a similar check to see if between-group comparisons are allowed.

Reply:
I see, we still don't need a `useEffect` I think, switched to one of the custom selectors which takes care of most of it. Also added a `useMemo` for the part that performs the `length === 1` check to reduce the amount of re-renders

Reply:
sounds good -- I'll leave the comment unresolved just as a reminder to myself to check that batch DE works for Seurat object uploads

Reply:
works nicely :)